sdk/core/azure-core/src/test/java/com/azure/core/http/policy/RedirectPolicyTest.java

```diff
@@ -326,6 +327,32 @@ public void nonRedirectRequest() throws MalformedURLException {
         assertEquals(401, response.getStatusCode());
     }
 
+    @Test
+    public void defaultRedirectAuthorizationHeaderCleared() throws Exception {
+        RecordingHttpClient httpClient = new RecordingHttpClient(request -> {
+            if (request.getUrl().toString().equals("http://localhost/")) {
+                Map<String, String> headers = new HashMap<>();
+                headers.put("Location", "http://redirecthost/");
+                headers.put("Authorization", "12345");
+                HttpHeaders httpHeader = new HttpHeaders(headers);
+                return Mono.just(new MockHttpResponse(request, 308, httpHeader));
```

Comment:
Should we test it for both 307 and 308 as both are our default redirect strategy return codes?